src/twisted/test/test_defer.py

```diff
@@ -3557,3 +3558,96 @@ async def testFunction() -> bool:
         clock.advance(1)
 
         self.assertEqual(self.successResultOf(d), True)
+
+
+class InlineCallbacksExceptionContextTests(unittest.TestCase):
+    @defer.inlineCallbacks
+    def doNothingReturnValue(self) -> Generator[Any, Any, None]:
+        """
+        L{defer.inlineCallbacks}-decorated function that returns using _DefGen_Return.
+        """
+        yield None
+        defer.returnValue("doNothingReturnValue")
+
+    @defer.inlineCallbacks
+    def doNothingReturn(self) -> Generator[Any, Any, Any]:
+        """
+        L{defer.inlineCallbacks}-decorated function that returns using StopIteration.
+        """
+        yield None
+        return "doNothingReturn"
+
+    @defer.inlineCallbacks
+    def doNothingRaise(self) -> Generator[Any, Any, None]:
+        yield None
+        raise Exception("Errback exception")
+
+    @defer.inlineCallbacks
+    def test_exceptionContextEmtpyInCallback(
+        self,
+    ) -> Generator[Deferred[Any], Any, None]:
+        """
+        When in the callbacks of an L{defer.inlineCallbacks}-decorated function
+        there should not be any exception context if there wasn't any to begin
+        with.
+        """
+        for do_nothing in (self.doNothingReturn, self.doNothingReturnValue):
+            yield do_nothing()
+            # There were never any exceptions that this function was aware of
+            # (_DefGen_Return and StopIteration are implementation details
+            #  of 'do_nothing' and shouldn't leak into the exc_info of the
+            #  caller)
+            self.assertEqual(sys.exc_info(), (None, None, None))
```

Comment:
Ah amazing I've always seen weird StopIteration related Tracebacks and had no idea why